Tools/Scripts/webkitpy/featuredefines/search.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (C) 2022 Sony Interactive Entertainment
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Searches files for feature defines."""
+
+import fnmatch
+import os
+import re
+
+from abc import abstractmethod, ABCMeta
+from io import open
+from webkitpy.featuredefines.matcher import flag_matcher, usage_matcher, idl_usage_matcher, cmake_options_matcher, declaration_matcher
+
+
+class FeatureDefinesSearch(object):
+    """ Base class for searching for feature defines. """
+
+    ___metaclass___ = ABCMeta
```

Comment:
I'm a bit surprised unit tests passed with this? This is a Python 3 specific idiom, if memory serves. It think it's fine for us to use this, but we should probably disable `webkitpy/featuredefines` in our Python 2 unit tests, and probably add an assert in the `__init__` or these `.py` files when they contain something Python 3 specific.

(as a note, there is intention to totally purge Python 2 from our codebase, but that's not high on our priority list at the moment)

Reply:
There aren't any associated tests for this file so maybe that's why?

Reply:
Are there examples on how to mark something as Python 3 only in webkitpy?

Reply:
> Are there examples on how to mark something as Python 3 only in webkitpy?

There are not, but the gist of it would be to add `tester.skip` in `Tools/Scripts/webkitpy/test/main.py` and then do something like:
```
if sys.version_info < (3, 0):
    raise NotImplementedError
```

Reply:
Actually looks like itd be ok https://python.readthedocs.io/en/v2.7.2/library/abc.html